clang/lib/CodeGen/CGCUDANV.cpp

```diff
@@ -1160,9 +1152,8 @@ void CGNVCUDARuntime::createOffloadingEntries() {
 
 // Returns module constructor to be added.
 llvm::Function *CGNVCUDARuntime::finalizeModule() {
+  transformManagedVars();
```

Comment:
This does not look like "NFC" as we now perform the transform for the host compilation, too.

I assume we do have existing tests covering generation of the variables.

Reply:
I'm guessing we also don't have a test for `__managed__` in the external test suite?

Reply:
we did the equivalent transformation previously during registration of managed var by calling replaceManagedVar directly, so it is actually NFC. We have lit test https://github.com/llvm/llvm-project/blob/main/clang/test/CodeGenCUDA/managed-var.cu that covers transformation in both device and host code.

We also have execution tests in hip-tests https://github.com/llvm/llvm-project/blob/main/clang/test/CodeGenCUDA/managed-var.cu

Reply:
we do not have test for managed var in llvm-test-suite